### ISSUE TITLE
Add ability to match on entire html

### ISSUE THREAD

**Issue by srcrip** [germsvel/phoenix_test#85]
To retain the ability to match on html instead of a selector, you can use a helper function like this:

```
  def assert_html(session, text) do
    assert render(session.view) =~ text

    session
  end
```

I think this should probably just be added in as it's own function in the Assertions module.

**Comment by germsvel** [comment]
Yeah, I'm not opposed to the library adding something like that. I think I would name it `assert_text` since we're asserting there's some text on the page. I'd be happy to review a PR If you're up for it.

**Comment by srcrip** [comment]
I think `assert_html` may be preferable because `assert_text` implies plain text, ie, the html tags stripped away

**Comment by germsvel** [comment]
That's an interesting thought. I think we read it from different angles (which is always interesting!) 

This is where my head's at 👇 

As a user of the library, I would expect `assert_html` would help me assert that there's some HTML on the page:

```elixir
assert_html(session, "<div>Could not find user</div>")
``` 

But my guess is that people want to use it as an _"assert this text is on the page"_: 

```elixir
assert_text(session, "Could not find user")
```

I think that's also evidenced by the implementation you suggested -- where you have `text` as the argument being passed in (not `html`). 

And if instead we _wanted_ people to assert on HTML structure, I'm not sure I would recommend that. I typically recommend the opposite because asserting on HTML structure tends to make the tests brittle. For example, what if I add a class or id to my HTML in the first example? I've broken my test for no reason. 

If we want to assert on certain aspects of HTML, we should use `assert_has` since it lets you target things with CSS selectors.  

Finally, I think `assert_text` keeps in line with things like Wallaby's [`assert_text`](https://hexdocs.pm/wallaby/Wallaby.Browser.html#assert_text/2). We're _not_ trying to keep in line with that, of course, but I do think `assert_text` might be what people _expect_ to exist because of previous tools they've used.

**Comment by germsvel** [comment]
Looping back on this, we can introduce an `assert_text/2` if people still want it. 

But I think if we do, we should make it so assertions don't fail on apostrophes (or other characters like that). That's one of the problems with `html_string =~ "some text"` assertions: 

```elixir
assert html =~ "user's email"
# this ^ fails because of the apostrophe

assert html =~ "user&#39;s email"
# this succeeds
```

There is no world in which I want to write the second assertion. So it'd be nice if we could make `assert_text(session, "user's email")` work.

**Comment by srcrip** [comment]
That should be pretty doable by having it use that Plug method under the hood that does html escaping

**Comment by germsvel** [comment]
@srcrip that sounds good. If you (or someone else is interested in contributing this), I'd be happy to review a PR!

**Comment by germsvel** [comment]
Sorry @siraj-samsudeen . I missed your message. The link is now broken (you might've deleted the repo by now). If you or someone else is interested in opening a PR, happy to review it.
